include/service/entities/site_object_service.js

```diff
@@ -0,0 +1,134 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var async = require('async');
+
+module.exports = function (pb) {
+  //pb dependencies
+  var util = pb.util;
+  var BaseObjectService = pb.BaseObjectService;
+  var SiteService = pb.SiteService;
+
+  /**
+   * SiteObjectService
+   * @class SiteObjectService
+   * @constructor
+   * @extends BaseObjectService
+   */
+  function SiteObjectService(context) {
+    if (!util.isObject(context)) {
+      context = {};
+    }
+
+    context.type = TYPE;
+    SiteObjectService.super_.call(this, context);
+    this.siteService = new SiteService();
```

Comment:
Why didn't you just make the current SiteService extend from BaseObjectService?  Now there is an additional prototype that does almost the same thing and makes it more confusing as to which one should be used.